src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java

```diff
@@ -87,13 +88,26 @@
         return create(status, throwable, request, new HttpHeaders());
     }
 
+    default ResponseEntity<Problem> create (final HttpStatus status, final Throwable throwable,
```

Comment:
If we go for the `convert` method as part of `AdviceTrait` I'd suggest to change the implementation to

```java
return create(convert(status), throwable, request);
```

I.e. delegate to the method above. In addition I'd switch places with the method above, because we try to structure the code from abstract (top) to concrete (bottom), i.e. the methods you depend on are found further down below.